volume/mounts/windows_parser.go

```diff
@@ -246,8 +247,22 @@ func (p *windowsParser) validateMountConfigReg(mnt *mount.Mount, additionalValid
 		if err != nil {
 			return &errMountConfig{mnt, err}
 		}
+
+		if windowsDetectMountType(mnt.Target) == mount.TypeNamedPipe {
```

Comment:
Added this for the following test case: 
https://github.com/moby/moby/blob/7a44e4cde0b85c81642aac69f7a5d5fe76ea86f4/volume/mounts/windows_parser_test.go#L133

Earlier, this case failed the mounting because a source directory wasn't created. But, we can't mount a directory to a named pipe but always allowed mounting to volumes and directories.